src/sgame/sg_missile.cpp

```diff
@@ -332,20 +333,20 @@ static int ImpactHive( gentity_t *ent, trace_t*, gentity_t *hitEnt )
 	}
 }
 
-static int DefaultImpactFunc( gentity_t*, trace_t*, gentity_t* )
+static int DefaultImpactFunc( gentity_t*, const trace2_t*, gentity_t* )
 {
 	return MIB_IMPACT;
 }
 
-static void MissileImpact( gentity_t *ent, trace_t *trace )
+static void MissileImpact( gentity_t *ent, const trace2_t *trace )
 {
 	int       dirAsByte, impactFlags;
 	const missileAttributes_t *ma = BG_Missile( ent->s.modelindex );
 	gentity_t *hitEnt   = &g_entities[ trace->entityNum ];
 	gentity_t *attacker = &g_entities[ ent->r.ownerNum ];
 
 	// Returns whether damage and hit effects should be done and played.
-	std::function<int(gentity_t*, trace_t*, gentity_t*)> impactFunc;
+	int (*impactFunc)( gentity_t *, const trace2_t *, gentity_t * );
```

Comment:
why go back to C funcs?

Reply:
My instinct was to use the most specific type possible so that it is more efficient. Doesn't really matter though

Hopefully I will get the impact stuff as a CBSE message